librz/util/str.c

```diff
@@ -3510,30 +3510,31 @@ RZ_API wchar_t *rz_str_mb_to_wc_l(const char *buf, int len) {
 
 RZ_API char *rz_str_wc_to_mb_l(const wchar_t *buf, int len) {
 	char *res_buf = NULL;
-	bool fail = true;
 	size_t sz;
+	mbstate_t mbs;
+	const wchar_t *tmp = buf;
 
 	if (!buf || len <= 0) {
 		return NULL;
 	}
-	sz = wcstombs(NULL, buf, len);
+	mbrlen(NULL, 0, &mbs);
+	sz = wcsrtombs(NULL, &tmp, 0, &mbs);
```

Comment:
I don't think that's a correct use of mbrlen. From what I understood by reading docs mbrlen requires an initialized mbstate_t structure. Also what's the point of using `wcsrtombs` instead of wcstombs if you are not using the restartability feature?

Reply:
`A call to the function with a null pointer as pmb resets the shift state (and ignores parameter max).`
http://www.cplusplus.com/reference/cwchar/mbrlen/

Reply:
`wcsrtombs` allows to pass a null pointer to get the size (like snprintf). The issue is that you can't use `wcstombs` with null ptr.

Reply:
cplusplus.com is only place suggesting such use of mbrlen.  Cppreference and microsoft documentation doesn't mention such behavior and suggest to zero initialize the mbstate_t structure instead.

Why do you think wcstombs can't be used with null ptr for such use case. From what I understood GCC warning was because it was used with `dst=NULL` and `len > 0` at the same time not because using NULL in general.

Reply:
On the other hand we might want to use wcsrtombs in both calls, because nonrestartable version internally uses a global mbstate_t variable making it not thread safe.

Reply:
i actually tried to think about this. i wasn't sure because the references are mixed; but the behavior of wcsrtombs is standardized.
therefore i do not know. i can close this and let somebody else fix this.

Reply:
@karliss after reading more, i decided to go your way.